src/web/page/pages.rs

```diff
@@ -0,0 +1,64 @@
+use super::TEMPLATE_DATA;
+use crate::docbuilder::Limits;
+use iron::{headers::ContentType, response::Response, status::Status, IronResult};
+use serde::Serialize;
+use tera::Context;
+
+/// The central trait that rendering pages revolves around, it handles selecting and rendering the template
+pub trait WebPage: Serialize + Sized {
+    /// Turn the current instance into a `Response`, ready to be served
+    // TODO: We could cache similar pages using the `&Context`
+    fn into_response(self) -> IronResult<Response> {
+        let ctx = Context::from_serialize(&self).unwrap();
+        let rendered = TEMPLATE_DATA.templates.load()
+            .render(Self::template(), &ctx)
+            .unwrap();
+
+        let mut response = Response::with((self.get_status(), rendered));
+        response.headers.set(Self::content_type());
+
+        Ok(response)
+    }
+
+    /// Get the name of the template to be rendered
+    fn template() -> &'static str;
+
+    /// Gets the status of the request, defaults to `Ok`
+    fn get_status(&self) -> Status {
+        Status::Ok
+    }
+
+    /// The content type that the template should be served with, defaults to html
+    fn content_type() -> ContentType {
+        ContentType::html()
+    }
+}
+
+/// The sitemap, corresponds with `templates/sitemap.xml`
```

Comment:
I'd prefer if those structs were defined with the rest of that page's content (for example, this struct should be in `src/web/sitemap.rs`.

Reply:
I had it like that originally, but it was a lot easier to reason about and cleaner to have all the stuff that only applies to rendering all grouped together

Reply:
Do you think it's cleaner when tweaking with the rendering of the whole website or that it's cleaner when working on a single page?

Reply:
Both in my opinion, it means all the cognitively similar things are located in the same place

Reply:
Hmm, I slightly prefer having everything related to a page in that page's file. @jyn514 @Nemo157  what's your opinion here?

Reply:
I agree with having everything in the same file, that lets you see what custom variables have been added for that page without having to switch between files.